MobileSyncExplorerKotlinTemplate/app/src/main/java/com/salesforce/mobilesyncexplorerkotlintemplate/contacts/listcomponent/ui/ContactsListContentCore.kt

```diff
@@ -0,0 +1,235 @@
+import com.salesforce.mobilesyncexplorerkotlintemplate.model.contacts.ContactObject
+
+@Composable
+fun ContactsListContent(
+    modifier: Modifier = Modifier,
+    uiState: ContactsListUiState,
+    listClickHandler: ContactsListClickHandler,
+) {
+    SubcomposeLayout(modifier = modifier) { constraints ->
+        val searchConstraints = constraints.copy(minWidth = 0, minHeight = 0)
+
+        val searchBarMeasureables = subcompose(slotId = ID_FLOATING_SEARCH_BAR) {
+            SearchBar(
+                modifier = Modifier.layoutId(ID_FLOATING_SEARCH_BAR),
+                isSearchActive = uiState.isSearchJobRunning,
+                field = uiState.searchField
+            )
+        }
+
+        val searchBarPlaceable = searchBarMeasureables
+            .first { it.layoutId == ID_FLOATING_SEARCH_BAR }
+            .measure(searchConstraints)
+
+        val listMeasureables = subcompose(slotId = ID_CONTACTS_LIST) {
+            val paddingDp = with(LocalDensity.current) { searchBarPlaceable.height.toDp() }
+
+            ListContentInner(
+                modifier = Modifier.layoutId(ID_CONTACTS_LIST),
+                contentPadding = PaddingValues(top = paddingDp),
+                uiState = uiState,
+                listClickHandler = listClickHandler
+            )
+        }
+
+        val listPlaceable = listMeasureables
+            .first { it.layoutId == ID_CONTACTS_LIST }
+            .measure(constraints)
+
+        layout(
+            width = maxOf(searchBarPlaceable.width, listPlaceable.width),
+            height = maxOf(searchBarPlaceable.height, listPlaceable.height)
+        ) {
+            listPlaceable.placeRelative(0, 0)
+            searchBarPlaceable.placeRelative(0, 0)
+        }
+    }
+}
+
+@Composable
+private fun ListContentInner(
+    modifier: Modifier = Modifier,
+    contentPadding: PaddingValues,
+    uiState: ContactsListUiState,
+    listClickHandler: ContactsListClickHandler
+) {
+    LazyColumn(
+        modifier = modifier,
+        contentPadding = contentPadding
+    ) {
+        items(items = uiState.contacts, key = { it.id }) { record ->
+            ContactCard(
+                modifier = Modifier.padding(4.dp),
+                startExpanded = false,
+                model = record.sObject,
+                syncState = record.localStatus.toUiSyncState(),
+                onCardClick = { listClickHandler.contactClick(record.id) },
+                onDeleteClick = { listClickHandler.deleteClick(record.id) },
+                onUndeleteClick = { listClickHandler.undeleteClick(record.id) },
+                onEditClick = { listClickHandler.editClick(record.id) },
+            )
+        }
+    }
+}
+
+@Composable
+private fun SearchBar(
+    modifier: Modifier = Modifier,
+    isSearchActive: Boolean,
+    field: EditableTextFieldUiState
+) {
+    FloatingTextEntryBar(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(8.dp),
```

Comment:
I am not super up to date on compose style, but should things like padding be a constant somewhere?  `8.dp` is still a magic number right?